anypoint-mq/3.x/modules/ROOT/pages/anypoint-mq-failover.adoc

```diff
@@ -0,0 +1,118 @@
+
+[IMPORTANT]
+When you enable failover for a queue and the primary region goes down,
+your data resides in a different region.
+
+
+To use the failover feature, do the following:
+
+* In your app:
+
+.. Upgrade Anypoint MQ Connector to version 3.2.6 or later.
+.. <<enable-failover-feature,Enable failover in your app.>>
+.. (Optional) <<configure-check-frequency,Change the check frequencies>>. 
+
+* In Anypoint MQ:
+.. Create or modify a queue and xref:mq::mq-failover.adoc#enable-failover[enable failover on the queue].
+
+[[enable-failover-feature]]
+== Enable Failover in Your App
+
+To enable the failover feature in your app, set the `cross.region.failover.feature.enabled` system property to `true`.
+
+To set this application property in Runtime Manager:
+
+. Sign in to Anypoint Platform and click *Runtime Manager* in the navigation menu.
+. Deploy a new app or change the settings for a deployed app.
+. On the *Properties* tab, specify values for the property:
++
+image::mq::mq-set-properties-failover-enabled.png["Properties tab with failover properties specified"]
+
+For more information, see: 
+
+* xref:cloudhub-2::ch2-manage-props.adoc[Changing App Behavior with Properties on CloudHub 2.0]
+* xref:cloudhub::cloudhub-manage-props.adoc[]
+
+Alternatively, set the property in *Run Configurations* in Anypoint Studio.
+For more information, see xref:mule-runtime::mule-app-properties-system.adoc[System Properties in Mule Apps] 
+
+
+
+[[configure-check-frequency]]
+== Configure the Frequency of Fallback Queue Checks
+
+By default, Anypoint MQ Connector checks the fallback queue once per minute.
+To change this behavior, set the following properties:
+
+* `fallback.messages.status.check.interval.ms`
++
+Specifies how often the connector checks the fallback queue for any messages received.
+* `primary.region.status.check.interval.ms`
++
+Specifies how often the connector checks that the primary region is up when processing from the fallback region.
+
+To set these application properties in Runtime Manager:
+
+. Sign in to Anypoint Platform and click *Runtime Manager* in the navigation menu.
+. Deploy a new app or change the settings for a deployed app.
+. On the *Properties* tab, specify values for the following properties (in seconds):
++
+image::mq::mq-set-properties-failover.png["Properties tab with failover properties specified"]
+
+For more information, see: 
+
+* xref:cloudhub-2::ch2-manage-props.adoc[Changing App Behavior with Properties on CloudHub 2.0]
+* xref:cloudhub::cloudhub-manage-props.adoc[]
+
+Alternatively, set these properties in *Run Configurations* in Anypoint Studio.
+For more information, see xref:mule-runtime::mule-app-properties-system.adoc[System Properties in Mule Apps] 
+
+[[check-interval]]
+=== Check Interval Considerations
+
+To determine how to configure the application property values: 
+
+* When publishing to the fallback region, you might experience latencies because every publish check if the primary region is up.
```

Comment:
When publishing to the fallback region during primary region downtime, the app might experience latencies because the publish operation in MQ Connector tries to check if the primary region is back up every 1 minute by default. 

If the publish operation in a flow is triggered less frequently than 1 minute, there will be a check to see if primary region is back up every time the operation is triggered and this introduces latency.

To reduce publish latency, set `primary.region.status.check.interval.ms` to be greater than the frequency at which publish operation is triggered in your flow.